lib/apiservers/portlayer/restapi/handlers/containers_handlers.go

```diff
@@ -358,20 +358,15 @@ func (handler *ContainersHandlersImpl) ContainerWaitHandler(params containers.Co
 		})
 	}
 
-	// if the container is already stopped return
-	if c.State != exec.StateRunning {
+	select {
+	case <-c.WaitForState(exec.StateStopped):
```

Comment:
The rest handlers are purely about exposing the RPC - we shouldn't be packing additional logic into them.

Is there a reason not to put the select clause into exec.Wait?